src/main/java/org/java_websocket/server/WebSocketServer.java

```diff
@@ -721,6 +721,9 @@ public final void onWebsocketOpen(WebSocket conn, Handshakedata handshake) {
 
   @Override
   public final void onWebsocketClose(WebSocket conn, int code, String reason, boolean remote) {
+    // fix: reason display suppressed by removeConnection(conn) or onClose(...)
+	  if (reason != null)                      // added
+		log.error("Reason: " + reason);   // added
```

Comment:
The reason is already in the call to onClose()

Reply:
That's true, Marcel, but it wasn't being displayed!  Instead, `onClose()` was displaying a "close" message, so I had no idea what was really causing the close!  There is probably a better way to make sure `reason` is displayed, but I was in a hurry!  TIA